src/ServiceInsight/ServiceControl/DefaultServiceControl.cs

```diff
@@ -537,6 +557,7 @@ void LogError(IRestResponse response)
         static bool HasSucceeded(IRestResponse response) => successCodes.Any(x => response != null && x == response.StatusCode && response.ErrorException == null);
 
         static IEnumerable<HttpStatusCode> successCodes = new[] { HttpStatusCode.OK, HttpStatusCode.Accepted, HttpStatusCode.NotModified, HttpStatusCode.NoContent };
+        static Regex linkExpression = new Regex(@"<([^>]+)>;\s?rel=""(\w+)"",?\s?", RegexOptions.Compiled);
```

Comment:
_Continuing from https://github.com/Particular/ServiceInsight/pull/737/files#r133940544_

OK, let's rewind: can you explain how the regex works?

Reply:
I'm not able to make sense of the regex and would much rather prefer an iterative approach.

Is it sufficient to split on the first `,` that follows a `>;`? Or are there cases where there won't be any `>`'s?

Or what about splitting on '<'?

Reply:
This is how the links look like

![image](https://user-images.githubusercontent.com/174258/35396793-f2aee0f4-01ed-11e8-9f24-3e0d407c741a.png)

Reply:
Is the regex really a problem here? I mean string splitting can easily get as complex as understanding a regex.

Reply:
The splitting would be based on the context (we can't just split at `;` because it may be a part of a URL. I am glad that the SI maintainers accepted this solution. I would love to see this simplified.